### PR TITLE
Update Angular, React and HTML Toggle components

### LINKED ISSUE #2944: [Enhance] Toggle's link will be a <button> - Angular
**User Story:**
As Spark we want Toggle to use `<button>` to open and close so that we are accessible. 

**Notes:**
Current State: Toggle uses a link to open and close it, since it doesn't navigate, it should be a button. 
Requestor: A11y audit, Section RM-12.

**AC:**
Angular Toggle
1. Angular Toggle will replace  `<a>` with `<button>` and fix the CSS.
2. Will see aria-controls. (Bob note: This can be added to the component with a lodash call if the ID doesn't exist.)

aria-controls this attribute is on the trigger element (the button). It's value is set to the id of the content element. If the user provides an id use that. If the user doesn't provide an id, generate one with lodash.

```
<button aria-controls="123"></button>
<div class="im_the_content" id="123">The content</div>`

### PR DESCRIPTION
**Pull request by guern1kn**
## What does this PR do?
- [x] Update the Angular Toggle component to use `<button>` instead of `<a>`
- [x] Add `aria-controls` to the Angular Toggle component
- [x] Update tests for Angular Toggle component
- [x] Update `generateAriaControls` function and related stories
- [x] Update _toggle.scss to remove font styles from `sprk-c-Toggle__trigger`
- [x] Update HTML story for toggle to use the font style class and `sprk-u-TextCrop--none`
- [x] Update React component for toggle to use the correct classes

### Associated Issue
Fixes #2944 

### Code
 - [x] Build Component in Angular
 - [x] Build Component in React
 - [x] Build Component in HTML
 - [x] Unit Testing in Angular with `npm run coverage` in `angular/` (100% coverage, 100% passing)
 - [x] Unit Testing in React with `npm run test` in `react/` (100% coverage, 100% passing)
 - [x] Unit Testing in HTML with `npm run coverage` in `html/` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
Tested all three frameworks in each browser since there were changes to all.
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### Screenshots
image

### CODE REVIEW

File: angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.spec.ts [hunk 1]
```diff
@@ -86,4 +90,52 @@ describe('SprkToggleComponent', () => {
...
+    contentElement.setAttribute('id', 'foo');
+    expect(triggerElement.hasAttribute('aria-controls')).toEqual(false);
+    fixture.detectChanges();
+
+    expect(triggerElement.hasAttribute('aria-controls')).toEqual(true);
+    expect(triggerElement.getAttribute('aria-controls')).toEqual('foo');
+    expect(contentElement.getAttribute('id')).toEqual('foo');
+  });
+
+  it('should not change values if aria-controls and are both present but have different values', () => {
```
Comment (afebbraro) [inline angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.spec.ts]: '
```suggestion
  it('should not change values if aria-controls and id are both present but have different values', () => {
```

File: angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.spec.ts [hunk 2]
```diff
@@ -86,4 +90,52 @@ describe('SprkToggleComponent', () => {
...
+    triggerElement.setAttribute('aria-controls', 'foo');
+    contentElement.setAttribute('id', 'bar');
+    fixture.detectChanges();
+
+    expect(triggerElement.getAttribute('aria-controls')).toEqual('foo');
+    expect(contentElement.getAttribute('id')).toEqual('bar');
+  });
+
+  it('should not change values if aria-controls and are both present and have correct values', () => {
```
Comment (afebbraro) [inline angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.spec.ts]:
```suggestion
  it('should not change values if aria-controls and id are both present and have correct values', () => {
```

File: angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts [hunk 3]
```diff
@@ -121,12 +126,46 @@ export class SprkToggleComponent implements OnInit {
...
+    // set the value of aria-controls
+    triggerElement.setAttribute('aria-controls', contentId);
+  }
+
   ngOnInit() {
     this.toggleState();
+
+    // Get the toggle's trigger and content elements
+    const toggleTrigger = document.querySelector('[data-sprk-toggle="trigger"]');
+    const toggleContent = document.querySelector('[data-sprk-toggle="content"]');
```
Comment (afebbraro) [inline angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts]: we should be doing this the Angular way.

Comment (guern1kn) [reply angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts]: ` const toggleTrigger = this.ref.nativeElement.querySelector('[data-sprk-toggle="trigger"]');
  const toggleContent = this.ref.nativeElement.querySelector('[data-sprk-toggle="content"]');`

More like this? We are using these types of selectors for stepper as well

Comment (afebbraro) [reply angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts]: We should be setting attrs, adding classes etc with renderer2 (https://angular.io/api/core/Renderer2). And referencing elements inside the main component with viewchild https://www.angularjswiki.com/angular/how-to-select-an-element-in-angular-component-template/

File: angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts [hunk 4]
```diff
@@ -121,12 +128,46 @@ export class SprkToggleComponent implements OnInit {
...
+
+    // set the value of aria-controls
+    triggerElement.setAttribute('aria-controls', contentId);
+  }
+
   ngOnInit() {
     this.toggleState();
+
+    // Get the toggle's trigger and content elements
+    const toggleTrigger = this.ref.nativeElement.querySelector('[data-sprk-toggle="trigger"]');
```
Comment (afebbraro) [inline angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts]: We should be setting attrs, adding classes etc with renderer2 (https://angular.io/api/core/Renderer2). And referencing elements inside the main component with viewchild https://www.angularjswiki.com/angular/how-to-select-an-element-in-angular-component-template/

**Comment by afebbraro** [comment]
I'd like to see some tests that test out the logic you have for logging warnings.

File: html/components/_toggle.scss [hunk 5]
```diff
@@ -16,13 +16,6 @@
   text-decoration: $sprk-toggle-trigger-text-decoration;
   transition: $sprk-toggle-trigger-transition;
 
-  // Font considerations
-  color: $sprk-toggle-trigger-color;
```
Comment (afebbraro) [inline html/components/_toggle.scss:20]: Did you delete these out of settings too?

Comment (guern1kn) [reply html/components/_toggle.scss:20]: Nope! Sure didn't. I'll take care of that.

**Pull request merged**